cns/metric/configsnapshot.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018 Microsoft. All rights reserved.
+// MIT License
+
+package metric
+
+import (
+	"context"
+	"crypto/md5" //nolint:gosec // used for checksum
+	"encoding/json"
+	"time"
+
+	"github.com/Azure/azure-container-networking/aitelemetry"
+	"github.com/Azure/azure-container-networking/cns/configuration"
+	"github.com/Azure/azure-container-networking/cns/logger"
+	"github.com/pkg/errors"
+)
+
+// SendCNSConfigSnapshot emits CNS config periodically
+func SendCNSConfigSnapshot(ctx context.Context, config *configuration.CNSConfig) {
+	ticker := time.NewTicker(time.Minute * time.Duration(config.TelemetrySettings.ConfigSnapshotIntervalInMins))
+	defer ticker.Stop()
+
+	event, err := createCNSConfigSnapshotEvent(config)
+	if err != nil {
+		logger.Errorf("[Azure CNS] SendCNSConfigSnapshot: %v", err)
+		return
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
```

Comment:
this will only start emitting the first config snapshot one hour after cns starts. you may want to emit this right away as well.

Reply:
added to emit right away as well. we log the config on CNS start up, but doesn't hurt to emit it right away either